patroni/dcs/zookeeper.py

```diff
@@ -211,14 +239,14 @@ def touch_member(self, data, ttl=None):
 
         try:
             if member:
-                self._client.retry(self._client.set, path, data)
+                self._client.set_async(self.member_path, data).get(timeout=1)
             else:
-                self._client.retry(self._client.create, path, data, makepath=True, ephemeral=True)
+                self._client.create_async(self.member_path, data, makepath=True, ephemeral=True).get(timeout=1)
             self._my_member_data = data
             return True
         except NodeExistsError:
             try:
-                self._client.retry(self._client.set, path, data)
+                self._client.set_async(self.member_path, data).get(timeout=1)
```

Comment:
Seems that exception handler only belongs to the case of the failed create_async, we won't get this exception in the set_async case